doc/mgmt/SONiC-Management-Framework-Image-mgmt-HLD.md

```diff
@@ -0,0 +1,225 @@
+# Image Management
+Software upgrade and image installation support for SONiC
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev  |    Date    |      Author       | Change Description |
+| :--: | :--------: | :---------------: | ------------------ |
+| 0.1  | 09/13/2019 | Arunsundar Kannan | Initial version    |
+
+# About this Manual
+This document provides general information about the Image management and installation feature implementation in SONiC.
+# Scope
+Covers Northbound interface for the Image management feature, as well as Unit Test cases.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**         | **Meaning**                    |
+| ---------------- | ------------------------------ |
+| Image Management | Image installation, validation |
+
+# 1 Feature Overview
+
+Provide management framework capabilities to handle:
+
+- Image Installation
+- List available images
+- Removal of an image
+- Set default image for next boot
+
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+Provide management framework support to existing SONiC capabilities with respect to Image management
+
+### 1.1.2 Configuration and Management Requirements
+- CLI style configuration and show commands
+- REST API support
+- gNMI Support
+
+### 1.1.3 Scalability Requirements
+N/A
+### 1.1.4 Warm Boot Requirements
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+The feature will use the traditional translib/transformer approach with an addition of a service running in the host which does all the "actual work". The service will use the existing sonic installer script in the host.
```

Comment:
Define "actual work".
Describe more specifically what is meant by the "traditional translib/transformer approach." I think you'll also need to elaborate on RPC support and how that is supported by translib, or at least how transformer will handle translations from OpenConfig gNOI to operations on the container/host.